### PR TITLE
Reports-level scoping of stats engine

### PR DESCRIPTION
**Pull request by bttf**
This PR includes changes to allow users to set the **stats engine** for **experiment reports**

Background: We recently introduced the ability to configure a stats engine across an entire organization. However some users still need a way to pilot a different stats engine without changing it org-wide. This PR introduces an alternative to the org-wide setting - via an experiment report. This unlocks the ability to set stats engine in a contained context without changing it for all experiments in an org.

Changes
- Isolate `StatsEngine` enum type definition
- UI: Adds "Stats Engine" dropdown to Reports configuration page 
- Update `ReportInterface` to accept optional `statsEngine` property under `args` object
- Fix small bug that occurs in frequentist reports (packages/front-end/components/Experiment/ResultsDownloadButton.tsx)

Screenshots
image
image

### CODE REVIEW

**Comment by lukesonnet** [comment]
Minor q: Is there a way to easily add `(organization default)`  or `(project default)` next to the option in the Reports configuration page? (and if so, would we even want to add this level of business?)

**Comment by bttf** [comment]
> Minor q: Is there a way to easily add `(organization default)` or `(project default)` next to the option in the Reports configuration page? (and if so, would we even want to add this level of business?)

It would be easy. What you think about adding a 3rd option? (And later, a 4th for project-wide), e.g.:

- Bayesian
- Frequentist
- Organization Default (Bayesian)
- Project Default (Frequentist)

File: packages/front-end/components/Experiment/ResultsDownloadButton.tsx
Comment (lukesonnet) [inline packages/front-end/components/Experiment/ResultsDownloadButton.tsx:91]: Wait, what's this bug (I introduced it)? Does it throw some error if `ci` is missing that I missed during my testing?

**Review by lukesonnet — APPROVED** [review]
This lgtm, especially if you've tested the results appearing correctly with the correct settings.

IDK if you want someone who can review the ts a bit closer to check before landing.

File: packages/front-end/components/Experiment/ResultsDownloadButton.tsx
Comment (lukesonnet) [reply packages/front-end/components/Experiment/ResultsDownloadButton.tsx:91]: ahhhh in the console

**Pull request merged**
